packages/flutter/lib/src/rendering/semantics.dart

```diff
@@ -597,18 +623,34 @@ class SemanticsNode extends AbstractNode {
     int flags = _flags;
     int actions = _actions;
     String label = _label;
+    TextDirection textDirection = _textDirection;
     final Set<SemanticsTag> tags = new Set<SemanticsTag>.from(_tags);
 
     if (mergeAllDescendantsIntoThisNode) {
       _visitDescendants((SemanticsNode node) {
         flags |= node._flags;
         actions |= node._actions;
+        textDirection ??= node._textDirection;
         tags.addAll(node._tags);
         if (node.label.isNotEmpty) {
+          String nestedLabel = node.label;
+          if (textDirection != node.textDirection && node.textDirection != null) {
+            const String lre = '\u202A'; // U+202A LEFT-TO-RIGHT EMBEDDING
+            const String rle = '\u202B'; // U+202B RIGHT-TO-LEFT EMBEDDING
+            const String pdf = '\u202C'; // U+202C POP DIRECTIONAL FORMATTING
+            switch (node.textDirection) {
+              case TextDirection.ltr:
+                nestedLabel = '$lre$nestedLabel$pdf';
+                break;
+              case TextDirection.rtl:
+                nestedLabel = '$rle$nestedLabel$pdf';
```

Comment:
Crazy.  FWIW, I usually put the rtl case first because its first in the enum declaration.

Reply:
I flipped some around to be rtl-first where I noticed them.